src/main/java/org/generationcp/ibpworkbench/ui/programmethods/ProgramMethodsPresenter.java

```diff
@@ -298,11 +299,31 @@ public MethodView convertMethod(final Method method) {
 		return methodClasses;
 	}
 
+	private int getMprgn(final String mtype) {
+		int mprgn = 0;
+		switch(mtype) {
+			case "GEN":
```

Comment:
We should avoid duplication of constants such "GEN", "DER", "MAN". I know it's an existing practice but let's improve.
We can create a `MethodType` enum in Middleware with fields `code` and `name` and values:
`DERIVATIVE("DER", "Derivative"), GENERATIVE("GEN", "Generative"), MAINTENANCE("MAN", "Maintenance");`


Then we should use in relevant code - queries, view, logic (eg. in  ProgramMethodsView)

Reply:
Created enum for MethodType in Middleware, please view this [pr](https://github.com/IntegratedBreedingPlatform/Middleware/pull/789). Created enum (MethodType) used as replacement to constant string variables for method types.